AmplifyPlugins/Analytics/AWSPinpointAnalyticsPlugin/Dependency/Pinpoint/Endpoint/EndpointClient.swift

```diff
@@ -47,18 +52,24 @@ actor EndpointClient: EndpointClientBehaviour {
          pinpointClient: PinpointClientProtocol,
          archiver: AmplifyArchiverBehaviour = AmplifyArchiver(),
          currentDevice: Device = DeviceProvider.current,
-         userDefaults: UserDefaultsBehaviour = UserDefaults.standard) {
+         userDefaults: UserDefaultsBehaviour = UserDefaults.standard,
+         keychain: KeychainStoreBehavior = KeychainStore()
+    ) {
         self.configuration = configuration
         self.pinpointClient = pinpointClient
         self.archiver = archiver
         self.currentDevice = currentDevice
         self.userDefaults = userDefaults
+        self.keychain = keychain
 
-        if let attributes = userDefaults.object(forKey: Constants.attributesKey) as? [String: [String]] {
+        EndpointClient.migrateLegacyKeyValueStore(userDefaults, keychain, archiver)
```

Comment:
nit: You can use `Self.` instead of `EndpointClient.` and save some characters :P

Reply:
Updated to use `Self.`